src/OAuth2Form.tsx

```diff
@@ -1,384 +1,447 @@
-    PORTAL: 'https://portal-dev.dev.sagebase.org/',
+  REPO: "https://repo-dev.dev.sagebase.org/",
+  PORTAL: "https://portal-dev.dev.sagebase.org/",
 };
 
 (window as any).SRC = {
-    OVERRIDE_ENDPOINT_CONFIG: isStaging ? stagingConfig : isDev ? devConfig : prodConfig
-}
-
-export type OAuth2FormProps = {
-    context?: SynapseContextType
-}
-
-export const OAuth2Form = (props: OAuth2FormProps) => {
-    const [token, setToken] = useState<string>()
-    const [profile, setProfile] = useState<UserProfile>()
-    const [oidcRequestDescription, setOidcRequestDescription] = useState<OIDCAuthorizationRequestDescription>()
-    const [oauthClientInfo, setOauthClientInfo] = useState<OAuthClientPublic>()
-    const [redirectURL, setRedirectURL] = useState<string>()
-
-    const [isConsenting, setIsConsenting] = useState<boolean>(false)
-    const [isGettingUserProfile, setIsGettingUserProfile] = useState<boolean>(false)
-
-    const [error, setError] = useState<any>()
+  OVERRIDE_ENDPOINT_CONFIG: isStaging
+    ? stagingConfig
+    : isDev
+    ? devConfig
+    : prodConfig,
+};
 
-    const sendGTagEvent = (event: string) => {
-        // send event to Google Analytics
-        // (casting to 'any' type to get compile-time access to gtag())
-        const windowAny: any = window
-        const gtag = windowAny.gtag
-        if (gtag) {
-            gtag('event', event, {
-                'event_category': 'SynapseOAUTH',
-            })
-        }
+export const OAuth2Form = () => {
+  const { accessToken, setAccessToken } = useOAuthAppContext();
+  const [profile, setProfile] = useState<UserProfile>();
+  const [oidcRequestDescription, setOidcRequestDescription] =
+    useState<OIDCAuthorizationRequestDescription>();
+  const [oauthClientInfo, setOauthClientInfo] = useState<OAuthClientPublic>();
+  const [redirectURL, setRedirectURL] = useState<string>();
+
+  const [isConsenting, setIsConsenting] = useState<boolean>(false);
+  const [isGettingUserProfile, setIsGettingUserProfile] =
+    useState<boolean>(false);
+
+  const [error, setError] = useState<any>();
+
+  const sendGTagEvent = (event: string) => {
+    // send event to Google Analytics
+    // (casting to 'any' type to get compile-time access to gtag())
+    const windowAny: any = window;
+    const gtag = windowAny.gtag;
+    if (gtag) {
+      gtag("event", event, {
+        event_category: "SynapseOAUTH",
+      });
     }
-
-    const onError = (error: any) => {
-        debugger
-        handleErrorRedirect(error)
-        setError(error)
+  };
+
+  const onError = (error: Error | OAuthClientError | SynapseClientError) => {
+    debugger;
+    if (
+      error instanceof SynapseClientError &&
+      error.reason.includes("invalid_token")
+    ) {
+      // invalid token, so clear the profile, token, and cookie
+      SynapseClient.signOut(() => {
+        setAccessToken(undefined);
+        setProfile(undefined);
+      });
+    } else {
```

Comment:
This is the special case. If the token is invalid (in 99% of cases, expired), clear the token out. This will prompt the user to sign in again.